setup_script.sh

```diff
@@ -1,182 +1,159 @@
 #!/bin/bash
 
 ###
-## ensure that required params are passed into script
+## Ensure that required parameters are passed into the script
 ###
 if [ -n "$1" ]; then
     HOST=$1
 else
-    echo "[ database ] : name not passed, set to default value (127.0.0.1)"
+    echo "[ database ] : Host not passed, set to default value (127.0.0.1)"
     HOST="127.0.0.1"
 fi
 
 ###
-## specify which database to operate on, if not specified, set default to camic
+## Specify which database to operate on, if not specified, set default to camic
 ###
 if [ -n "$2" ]; then
     DB_NAME=$2
-    echo "[ database ] : name set as ${DB_NAME}"
+    echo "[ database ] : Database name set as ${DB_NAME}"
 else
-    echo "[ database ] : name not passed, set to default value (camic)"
+    echo "[ database ] : Database name not passed, set to default value (camic)"
     DB_NAME="camic"
 fi
 
 ###
-## check if the system has required services installed
+## Check if the system has required services installed
 ###
 if command -v "mongo" &>/dev/null; then
     MONGO="mongo"
 elif command -v "mongosh" &>/dev/null; then
     MONGO="mongosh"
 else
-    echo "mongo or mongo could not be found on path. Please ensure that mongo is installed and is on PATH"
+    echo "Neither mongo nor mongosh could be found. Please ensure that MongoDB is installed and available in the PATH."
+    exit 1
 fi
 
 if ! command -v "node" &>/dev/null; then
-    echo "node could not be found on path. Please ensure that node is installed and is on PATH"
-    exit
+    echo "Node.js could not be found on path. Please ensure that Node.js is installed and is on PATH."
+    exit 1
 fi
 
 ###
-# check for the existence of required files in the Distro and camicroscope repository
+# Check for the existence of required files in the caracal repository
 ###
-echo "Checking for required files"
-
-#if [[ ! -d ../Distro/db ]]
-#then echo "../Distro/db does not exist"
-#     exit
-#fi
+echo "Checking for required files..."
 
-if [[ ! -d ../Distro/config ]]
-then echo "../Distro/config does not exist"
-     exit
+if [[ ! -d ../caracal/config ]]; then
+    echo "../caracal/config does not exist"
+    exit 1
 fi
 
-if [[ ! -d ../Distro/jwt_keys ]]
-then echo "../Distro/jwt_keys does not exist"
-     exit
+if [[ ! -d ../caracal/jwt_keys ]]; then
+    echo "../caracal/jwt_keys does not exist"
+    exit 1
```

Comment:
I think you've misunderstood this context. This assumes you're deploying the caMicroscope stack without docker. So we assume you have relevant config from Distro.

This could be made more independent by copying/renaming the required files (e.g. make an empty jwt_keys dir, copy routes.json from an example, etc)

I'm not sure how useful this is in practice; I'm not aware of anyone deploying caracal this way currently.

Please let me know if you want help with this.

Reply:
Would be grateful if you help me a bit.I am new to open source contribution so I would appreciate your help very much